BlockchainSdk/Blockchains/Sui/SuiWalletManager.swift

```diff
@@ -0,0 +1,145 @@
+//
+// SuiWalletManager.swift
+// BlockchainSdk
+//
+// Created by Sergei Iakovlev on 28.08.2024
+// Copyright © 2024 Tangem AG. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class SuiWalletManager: BaseManager, WalletManager {
+    let networkService: SuiNetworkService
+    let transactionBuilder: SuiTransactionBuilder
+    
+    init(wallet: Wallet, networkService: SuiNetworkService, transactionBuilder: SuiTransactionBuilder) {
+        self.networkService = networkService
+        self.transactionBuilder = transactionBuilder
+        super.init(wallet: wallet)
+    }
+
+    override func update(completion: @escaping (Result<Void, any Error>) -> Void) {
+        cancellable = networkService.getBalance(address: wallet.address, coin: .sui, cursor: nil)
+            .sink(receiveCompletion: { [weak self] completionSubscriptions in
+                if case let .failure(error) = completionSubscriptions {
+                    self?.wallet.clearAmounts()
+                    completion(.failure(error))
+                }
+            }, receiveValue: { [weak self] coins in
+                self?.updateWallet(coins: coins)
+                completion(.success(()))
+            })
+    }
+    
+    func updateWallet(coins: [SuiGetCoins.Coin]) {
+        let objects = coins.compactMap({
+            SuiCoinObject.from($0)
+        })
+        
+        let totalBalance = objects.reduce(into: Decimal(0)) { partialResult, coin in
+            partialResult += coin.balance
+        }
+        
+        let coinValue = totalBalance / wallet.blockchain.decimalValue
+        
+        if coinValue != wallet.amounts[.coin]?.value  {
+            wallet.clearPendingTransaction()
+        }
+        
+        wallet.add(coinValue: coinValue)
+        self.transactionBuilder.update(coins: objects)
+    }
+}
+
+extension SuiWalletManager: BlockchainDataProvider {
+    var currentHost: String {
+        networkService.host
+    }
+}
+
+extension SuiWalletManager: TransactionFeeProvider {
+    var allowsFeeSelection: Bool { false }
+    
+    func getFee(amount: Amount, destination: String) -> AnyPublisher<[Fee], any Error> {
+        networkService.getReferenceGasPrice()
```

Comment:
ну как и до этого и в других функциях то же самое
```suggestion
        return networkService.getReferenceGasPrice()
```